test/docfx.Test/restore/RestoreTest.cs

```diff
@@ -98,5 +98,23 @@ public static async Task RestoreUrls()
 
             Assert.Single(Directory.EnumerateFiles(restoreDir, "*"));
         }
+
+        [Theory]
+        [InlineData("abc123", "\"0xdef456\"", "abc123+etag+%220xdef456%22")]
+        [InlineData("abc123", "", "abc123+etag+")]
+        [InlineData("abc123", null, "abc123")]
+        public static void GetRestoreFileName(string hash, string etag, string expected)
+        {
+            Assert.Equal(expected, RestoreFile.GetRestoreFileName(hash, etag));
+        }
+
+        [Theory]
+        [InlineData("abc123+etag+%220xdef456%22", "\"0xdef456\"")]
+        [InlineData("abc123+etag+", "")]
```

Comment:
what is the purpose of `+etag+` here? is `abc123-%220xdef456%22` enough?

Reply:
`-` is not a safe separator as the encoded etag can also contain it. It can be `abc123+%220xdef456%22`, just insert `etag` to make it more readable.